src/Discord.Net.Rest/Entities/Messages/MessageHelper.cs

```diff
@@ -46,13 +46,39 @@ internal static class MessageHelper
             await client.ApiClient.RemoveAllReactionsAsync(msg.Channel.Id, msg.Id, options);
         }
 
-        public static async Task<IReadOnlyCollection<IUser>> GetReactionUsersAsync(IMessage msg, IEmote emote,
+        public static IAsyncEnumerable<IReadOnlyCollection<IUser>> GetReactionUsersAsync(IMessage msg, IEmote emote,
             Action<GetReactionUsersParams> func, BaseDiscordClient client, RequestOptions options)
         {
             var args = new GetReactionUsersParams();
             func(args);
-            string emoji = (emote is Emote e ? $"{e.Name}:{e.Id}" : emote.Name);
-            return (await client.ApiClient.GetReactionUsersAsync(msg.Channel.Id, msg.Id, emoji, args, options).ConfigureAwait(false)).Select(u => RestUser.Create(client, u)).ToImmutableArray();
+
+            var emoji = (emote is Emote e ? $"{e.Name}:{e.Id}" : emote.Name);
+            return new PagedAsyncEnumerable<IUser>(
+                DiscordConfig.MaxUsersPerBatch,
+                async (info, ct) =>
+                {
+                    if (info.Position != null)
+                        args.AfterUserId = info.Position.Value;
```

Comment:
The correct way imo would be creating a local "args" with the page size and the position.